src/Rel_PT_Emission Profile.jl

```diff
@@ -22,27 +22,48 @@ function D(R, f)
     return D
 end
 
+function (T(D))
+    T = (D/sigma_SB)^(1/4)
+    return T
+end
+
+function r_isco(a_star)
+
+    a = a_star
+    r_g = 2*G*M/(c^(2))
+    z1 = 1+(1-a^2)^(1/3)*((1+a)^(1/3)+(1-a)^(1/3))
+    z2 = (3*a^(2)+z1^(2))^(1/2)
+    if a >= 0
+        r_isco = (3+z2-(3-z1)*(3+z1+2*z2)^(1/2))*r_g
+    elseif a < 0
+        r_isco = (3+z2+(3-z1)*(3+z1+2*z2)^(1/2))*r_g
+    end
+
+    return r_isco
+end
+
 G = 6.67e-11
 c = 3e8
 L_sun = 3.8e26
 M_sun = 1.99e30
+sigma_SB = 5.67e-8
 
 eta = 0.1
-M = 10*M_sun
+M = 10e6*M_sun
 a_star = 0.5
-
-R_isco = 6*G*M/(c^2)
+R_isco = r_isco(a_star)
 L_edd = 3e4*L_sun*(M/M_sun)
-Mdot = L_edd/(c^2*eta)
+Mdot = -L_edd/(c^2*eta)
 mdot = 0.1*Mdot
 
 r_vals = LinRange(R_isco, 10*R_isco, 10000)
 D_vals = []
+T_vals = []
 
 for r in r_vals
     push!(D_vals, D(r, f(r)))
+    push!(T_vals, T(D(r, f(r))))
 end
```

Comment:
Julia has a very cool broadcast mechanism that lets you write this things cleaner, and will run faster (since output memory gets pre-allocated).

These chunk of code could be rewritten as
```julia
r_vals = LinRange(R_isco, 10*R_isco, 10000)

# . broadcasts a function element-wise over an array
f_r = f.(r_vals)

D_vals = D.(r_vals, f_r)
T_vals = T.(D_vals)
```

which is not only computationally faster, but a little easier to read.

Reply:
Thanks for this feedback, it all looks very helpful, and we'll try and tidy things up a bit using this.